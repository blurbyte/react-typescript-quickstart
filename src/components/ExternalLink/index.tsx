import styled from 'styled-components';

import { Color } from '@styles/colors';

const ExternalLink = styled.a.attrs({
  target: '_blank'
})`
  color: ${Color.Blue};
  cursor: pointer;
  text-decoration: none;
  outline: none;

  &:hover,
  &:focus {
    text-decoration: underline;
  }
`;

export default ExternalLink;
