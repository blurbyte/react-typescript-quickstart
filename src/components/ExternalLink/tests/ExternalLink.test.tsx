import * as React from 'react';
import { shallow } from 'enzyme';

import ExternalLink from '../index';

describe('ExternalLink', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ExternalLink href="http://taylorswift.com">Swift</ExternalLink>);
    expect(wrapper).toMatchSnapshot();
  });
});
