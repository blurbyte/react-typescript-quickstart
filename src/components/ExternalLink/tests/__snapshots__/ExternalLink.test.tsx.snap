// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink renders correctly 1`] = `
.c0 {
  color: #4a6cbb;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
  className="c0"
  href="http://taylorswift.com"
  target="_blank"
>
  Swift
</a>
`;
